go/vt/vtgate/planbuilder/routetree.go

```diff
@@ -220,66 +218,53 @@ func (rp *routeTree) searchForNewVindexes(ctx *planningContext, predicates []sql
 				Right:    &sqlparser.Subquery{Select: node.Subquery.Select},
 				Operator: originalCmp.Operator,
 			}
-			found, exitEarly, err := rp.planComparison(ctx, cmp)
-			if err != nil || exitEarly {
-				return false, err
+			found, exitEarly := rp.planComparison(ctx, cmp)
+			if exitEarly {
+				return false
 			}
 			newVindexFound = newVindexFound || found
 		case *sqlparser.ComparisonExpr:
-			found, exitEarly, err := rp.planComparison(ctx, node)
-			if err != nil || exitEarly {
-				return false, err
+			found, exitEarly := rp.planComparison(ctx, node)
+			if exitEarly {
+				return false
 			}
 			newVindexFound = newVindexFound || found
 		case *sqlparser.IsExpr:
-			found, err := rp.planIsExpr(ctx, node)
-			if err != nil {
-				return false, err
-			}
+			found := rp.planIsExpr(ctx, node)
 			newVindexFound = newVindexFound || found
 		}
 	}
-	return newVindexFound, nil
+	return newVindexFound
 }
 
-func (rp *routeTree) planComparison(ctx *planningContext, node *sqlparser.ComparisonExpr) (bool, bool, error) {
+func (rp *routeTree) planComparison(ctx *planningContext, node *sqlparser.ComparisonExpr) (bool, bool) {
```

Comment:
why do we want to remove the error here?

Reply:
the code inside the function is not throwing any error.